src/network.js

```diff
@@ -55,166 +57,211 @@ function networkRequest(options, oauthRequest) {
+
+    return new Promise((resolve, reject) => {
+      get(endpoint)
+        .then((messages) => {
+          if (messages.length > 0) {
+            // This call returned messages, add them to the list.
+            messagesList = messagesList.concat(messages);
+
+            if (messages.length < limit) {
+              // This call returned less than the number we asked for,
+              // which means we've reached the end of the messages.
+              // We don't have to recurse anymore.
+              resolve(messagesList);
+            }
+            else {
+              // We don't know if there are more messages or not,
+              // we must query for the next batch.
+              const lastMessageId = messages[messages.length - 1]['data']['name'];
+              resolve(getMessages(lastMessageId, messagesList));
+            }
+          }
+          else {
+            // We did not get any messages back from this query.
+            resolve(messagesList);
+          }
+        })
+        .catch(() => {
+          // There was an error executing this query.
+          resolve(messagesList);
+        });
+    });
   }
+
+  // Get all the messages from the inbox before proceeding.
+  getMessages()
+    .then((messages) => {
+      // Run filter logic against the entire list of messages.
+      messages
+        .filter(data => data['kind'] === 't4')
+        .map(data => data['data'])
+        .map (message => message['subject'].match(/^You\'ve been banned from participating in r\/(.+)$/i))
+        .filter(match => match !== null)
+        .map(match => match[1])
+        .forEach(subreddit => helper.log("- " + subreddit));
+    });
 }
 
 function getRedditComments(subreddit) {
-  let content = get("https://www.reddit.com/r/" + subreddit + "/comments.json?limit=100&raw_json=1");
-  if (!content) {
-    return;
-  }
-
-  let comments = content;
-
-  const lastProcessedIndex = comments.findIndex((el) => el['data']['name'] === lastProcessedCommentId);
-  if (lastProcessedIndex !== -1) {
-    comments = comments.slice(0, lastProcessedIndex)
-  }
-
-  const unprocessedComments = comments
-    .map(comment => comment['data'])
-    .map(data => {
-      return {
-          'body': data['body'],
-          'author': data['author'],
-          'id': data['name'],
-          'postTitle': data['link_title'],
-          'link': data['link_permalink'] + data['id'],
-          'subreddit': data['subreddit'],
-          'timestamp' : data['created_utc']
-        }
+  return get("https://www.reddit.com/r/" + subreddit + "/comments.json?limit=100&raw_json=1")
+    .then((comments) => {
+      if (!comments) {
+        return;
+      }
+
+      const lastProcessedIndex = comments.findIndex((el) => {
+        return el['data']['name'] === lastProcessedCommentId;
+      });
+      if (lastProcessedIndex !== -1) {
+        comments = comments.slice(lastProcessedIndex + 1, comments.length);
```

Comment:
I think this is what you wanted instead?

Previously it was
```js
comments = comments.slice(0, lastProcessedIndex)
```
, which will return all the comments _before_ the `lastProcessedIndex`.

This change,
```js
comments = comments.slice(lastProcessedIndex + 1, comments.length);
```
will return the comments _after_ the `lastProcessedIndex`.

For illustration:
```js
const lastIndex = 1;
const arr = ['a', 'b', 'c'];

// Previously
const slice1 = arr.slice(0, lastIndex); // returns ['a']

// Now
const slice2 = arr.slice(lastIndex + 1, arr.length); // returns ['c']
```

It seemed like we'd want the ones after the one we'd already seen. Maybe it works the other way, in which case this change should be reverted.

Reply:
Reddit's comment API returns the newest comments at the top of the array, see `created_utc` [here](https://www.reddit.com/r/all/comments.json?limit=100&raw_json=1), so do we want the ones before the id we have saved.

Reply:
Ok cool, I'll revert this then.

I tried not to change logic like this too often, and have left comments where I did.